### DISCUSSION TITLE
Create .sql file when a new change is committed in the database from sqlitestudio

### DISCUSSION THREAD

**Discussion by EduBic** [pawelsalawa/sqlitestudio#3913] [Q&A]
Hello

Is it possible to set an automatic generation of the .sql file (or call an external python script) when I change the .db file from SQLiteStudio?

If not, I saw that there is the possibility to create a plugin, can I have some suggestions about build external code triggered after the COMMIT event?

Thanks

**Comment by pawelsalawa** [comment]
If you want to track data changes (INSERT/UPDATE/DELETE) you can create triggers on all tables that will do `SELECT my_func();`.
Then you can create a custom SQL function named `my_func` that will do whatever you need. Including external file creation or Python script execution (Tcl language implementation for the function will allow you to do that).
